### PR TITLE
(CDAP-7106) Starts AuthorizationEnforcementService earlier to avoid u…

### PR DESCRIPTION
**Pull request by chtyim**
…nnecessary error message at startup time.

### CODE REVIEW

File: cdap-master/src/main/java/co/cask/cdap/data/runtime/main/TransactionServiceTwillRunnable.java [hunk 1]
```diff
@@ -111,10 +95,11 @@ protected void doInit(TwillContext context) {
 
   @Override
   public void getServices(List<? super Service> services) {
-    services.add(zkClient);
-    services.add(kafkaClient);
-    services.add(metricsCollectionService);
-    services.add(txService);
+    services.add(injector.getInstance(ZKClientService.class));
+    services.add(injector.getInstance(KafkaClientService.class));
+    services.add(injector.getInstance(AuthorizationEnforcementService.class));
```
Comment (gokulavasan) [inline cdap-master/src/main/java/co/cask/cdap/data/runtime/main/TransactionServiceTwillRunnable.java:100]: Do we need AuthorizationEnforcementService in TransactionServiceTwillRunnable now? Or is it for future usage?

File: cdap-watchdog/src/main/java/co/cask/cdap/metrics/runtime/MetricsProcessorTwillRunnable.java [hunk 2]
```diff
@@ -119,11 +112,12 @@ protected void doInit(TwillContext context) {
 
   @Override
   public void getServices(List<? super Service> services) {
-    services.add(zkClientService);
-    services.add(kafkaClientService);
+    services.add(injector.getInstance(ZKClientService.class));
+    services.add(injector.getInstance(KafkaClientService.class));
+    services.add(injector.getInstance(AuthorizationEnforcementService.class));
```
Comment (gokulavasan) [inline cdap-watchdog/src/main/java/co/cask/cdap/metrics/runtime/MetricsProcessorTwillRunnable.java:117]: same question as above

File: cdap-watchdog/src/main/java/co/cask/cdap/metrics/runtime/MetricsTwillRunnable.java [hunk 3]
```diff
@@ -103,10 +91,11 @@ protected void doInit(TwillContext context) {
 
   @Override
   public void getServices(List<? super Service> services) {
-    services.add(zkClient);
-    services.add(kafkaClient);
-    services.add(metricsCollectionService);
-    services.add(metricsQueryService);
+    services.add(injector.getInstance(ZKClientService.class));
+    services.add(injector.getInstance(KafkaClientService.class));
+    services.add(injector.getInstance(AuthorizationEnforcementService.class));
```
Comment (gokulavasan) [inline cdap-watchdog/src/main/java/co/cask/cdap/metrics/runtime/MetricsTwillRunnable.java:96]: same question as above

**Comment by gokulavasan** [comment]
One question. LGTM otherwise!

File: cdap-master/src/main/java/co/cask/cdap/data/runtime/main/TransactionServiceTwillRunnable.java (hunk 1 above)
Comment (chtyim) [inline cdap-master/src/main/java/co/cask/cdap/data/runtime/main/TransactionServiceTwillRunnable.java:100]: The problem is, it's part of the injector. This mean it is needed by some class, hence it's safer to always start it.

**Comment by chtyim** [comment]
Build passed: http://builds.cask.co/browse/CDAP-RUT119

**Comment by gokulavasan** [comment]
LGTM 👍

**Pull request merged**